test/memory/MemoryTestPublisher.cpp

```diff
@@ -0,0 +1,393 @@
+        PParam.rtps.builtin.use_SIMPLE_RTPSParticipantDiscoveryProtocol = true;
+        PParam.rtps.builtin.use_SIMPLE_EndpointDiscoveryProtocol = true;
+        PParam.rtps.builtin.m_simpleEDP.use_PublicationReaderANDSubscriptionWriter = true;
+        PParam.rtps.builtin.m_simpleEDP.use_PublicationWriterANDSubscriptionReader = true;
+        PParam.rtps.builtin.leaseDuration = c_TimeInfinite;
+        PParam.rtps.properties = part_property_policy;
+        PParam.rtps.sendSocketBufferSize = 65536;
+        PParam.rtps.listenSocketBufferSize = 2 * 65536;
+        PParam.rtps.setName("Participant_pub");
+        mp_participant = Domain::createParticipant(PParam);
+        if (mp_participant == nullptr)
+        {
+            return false;
+        }
+
+        Domain::registerType(mp_participant, (TopicDataType*)&memory_t);
+        Domain::registerType(mp_participant, (TopicDataType*)&command_t);
+
+        // DATA PUBLISHER
+        PublisherAttributes PubDataparam;
+        PubDataparam.topic.topicDataType = "MemoryType";
+        PubDataparam.topic.topicKind = NO_KEY;
+        std::ostringstream pt;
+        pt << "MemoryTest_";
+        if (hostname)
+            pt << asio::ip::host_name() << "_";
+        pt << pid << "_PUB2SUB";
+        PubDataparam.topic.topicName = pt.str();
+        PubDataparam.topic.historyQos.kind = KEEP_ALL_HISTORY_QOS;
+        PubDataparam.topic.historyQos.depth = n_samples;
+        PubDataparam.topic.resourceLimitsQos.max_samples = n_samples + 1;
+        PubDataparam.topic.resourceLimitsQos.allocated_samples = n_samples + 1;
+        PubDataparam.times.heartbeatPeriod.seconds = 0;
+        PubDataparam.times.heartbeatPeriod.fraction = 4294967 * 100;
+        if (!reliable)
+            PubDataparam.qos.m_reliability.kind = BEST_EFFORT_RELIABILITY_QOS;
+        Locator_t loc;
+        loc.port = 15000;
+        PubDataparam.unicastLocatorList.push_back(loc);
+        PubDataparam.properties = property_policy;
+        if (m_data_size > 60000)
+        {
+            PubDataparam.historyMemoryPolicy = eprosima::fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
+            PubDataparam.qos.m_publishMode.kind = eprosima::fastrtps::ASYNCHRONOUS_PUBLISH_MODE;
+        }
+        mp_datapub = Domain::createPublisher(mp_participant, PubDataparam, (PublisherListener*)&this->m_datapublistener);
+        if (mp_datapub == nullptr)
+            return false;
+
+        //COMMAND PUBLISHER
+        PublisherAttributes PubCommandParam;
+        PubCommandParam.topic.topicDataType = "TestCommandType";
+        PubCommandParam.topic.topicKind = NO_KEY;
+        std::ostringstream pct;
+        pct << "MemoryTest_Command_";
+        if (hostname)
+            pct << asio::ip::host_name() << "_";
+        pct << pid << "_PUB2SUB";
+        PubCommandParam.topic.topicName = pct.str();
+        PubCommandParam.topic.historyQos.kind = KEEP_ALL_HISTORY_QOS;
+        PubCommandParam.topic.historyQos.depth = 100;
```

Comment:
with keep_all, depth is not necessary. Remove it.

Reply:
This was added because of the bug related with the default depth, making the buffer to fill up and freezing the application.